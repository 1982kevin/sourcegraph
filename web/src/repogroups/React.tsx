import { RepogroupMetadata } from './types'
import { SearchPatternType } from '../../../shared/src/graphql/schema'
import * as React from 'react'
export const react: RepogroupMetadata = {
    title: 'React',
    name: 'react',
    url: '/react',
    description:
        "Explore React repositories on GitHub. The search examples show how to use React Hook useState with various data types.",
    examples: [
        {
            title: 'Find imports of useState with regex search',
            exampleQuery: <>import [^;]+useState[^;]+ from 'react'</>,
            rawQuery: "import [^;]+useState[^;]+ from 'react'",
            patternType: SearchPatternType.regexp,
        },
        {
            title: 'Examples of useState with an object as the input parameter',
            description: `useState takes a single argument and can accept a primitive, an array or an object.
            Syntax for usage is [state_value, function_to_update_state_value] = useState (initial_state_value).`,
            exampleQuery: <>{'useState({:[string]})'}</>,
            rawQuery: 'useState({:[string]}) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter',
            exampleQuery: <>useState(:[string])</>,
            rawQuery: 'useState(:[string]) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter in only TypeScript files',
            exampleQuery: (
                <>
                    useState(:[string])
                    <span className="search-keyword">lang:</span>typescript
                </>
            ),
            rawQuery: 'useState(:[string]) count:1000 lang:typescript',
            patternType: SearchPatternType.structural,
        },
    ],
    homepageDescription: 'Learn React with code examples.',
    homepageIcon: 'https://cdn4.iconfinder.com/data/icons/logos-3/600/React.js_logo-512.png',
}
