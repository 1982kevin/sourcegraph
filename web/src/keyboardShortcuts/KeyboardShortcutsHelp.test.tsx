import React from 'react'
import { KeyboardShortcutsHelp } from './KeyboardShortcutsHelp'
import { mount } from 'enzyme'

describe('KeyboardShortcutsHelp', () => {
    test('', () => {
        const output = mount(
            <KeyboardShortcutsHelp
                keyboardShortcuts={[
                    {
                        id: 'x',
                        title: 't',
                        keybindings: [{ held: ['Alt'], ordered: ['x'] }],
                    },
                ]}
                keyboardShortcutForShow={{
                    id: 'x',
                    title: 't',
                    keybindings: [{ held: ['Alt'], ordered: ['x'] }],
                }}
                forceIsOpen={true}
            />
        )
        expect(output).toMatchSnapshot()
    })
})
