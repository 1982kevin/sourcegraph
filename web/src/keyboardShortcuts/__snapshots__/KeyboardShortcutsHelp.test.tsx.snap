// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardShortcutsHelp  1`] = `
<KeyboardShortcutsHelp
  forceIsOpen={true}
  keyboardShortcutForShow={
    Object {
      "id": "x",
      "keybindings": Array [
        Object {
          "held": Array [
            "Alt",
          ],
          "ordered": Array [
            "x",
          ],
        },
      ],
      "title": "t",
    }
  }
  keyboardShortcuts={
    Array [
      Object {
        "id": "x",
        "keybindings": Array [
          Object {
            "held": Array [
              "Alt",
            ],
            "ordered": Array [
              "x",
            ],
          },
        ],
        "title": "t",
      },
    ]
  }
>
  <Shortcut
    held={
      Array [
        "Alt",
      ]
    }
    key="0"
    onMatch={[Function]}
    ordered={
      Array [
        "x",
      ]
    }
  >
    <ShortcutConsumer
      held={
        Array [
          "Alt",
        ]
      }
      onMatch={[Function]}
      ordered={
        Array [
          "x",
        ]
      }
    />
  </Shortcut>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={true}
    fade={false}
    isOpen={true}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    returnFocusAfterClose={true}
    role="dialog"
    scrollable={false}
    toggle={[Function]}
    unmountOnClose={true}
    zIndex={1050}
  >
    <Portal
      node={
        <div
          style="position: relative; z-index: 1050; display: block;"
          tabindex="-1"
        >
          <div
            class=""
          >
            <div
              class="modal"
              role="dialog"
              style="display: block;"
              tabindex="-1"
            >
              <div
                class="modal-dialog modal-dialog-centered"
                role="document"
              >
                <div
                  class="modal-content"
                >
                  <div
                    class="modal-header"
                  >
                    <h4
                      class="modal-title"
                    >
                      Keyboard shortcuts
                    </h4>
                    <button
                      aria-label="Close"
                      class="btn btn-icon"
                      data-dismiss="modal"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        ×
                      </span>
                    </button>
                  </div>
                  <div
                    class="modal-body"
                  >
                    <ul
                      class="list-group list-group-flush"
                    >
                      <li
                        class="list-group-item d-flex align-items-center justify-content-between"
                      >
                        t
                        <span>
                          <span>
                            <kbd>
                              Alt
                            </kbd>
                            <kbd>
                              x
                            </kbd>
                          </span>
                        </span>
                      </li>
                      <li
                        class="list-group-item d-flex align-items-center justify-content-between"
                      >
                        Expand URL to its canonical form (on file or tree page)
                        <span>
                          <span>
                            <kbd>
                              y
                            </kbd>
                          </span>
                        </span>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="modal-backdrop show"
            />
          </div>
        </div>
      }
    >
      <Portal
        containerInfo={
          <div
            style="position: relative; z-index: 1050; display: block;"
            tabindex="-1"
          >
            <div
              class=""
            >
              <div
                class="modal"
                role="dialog"
                style="display: block;"
                tabindex="-1"
              >
                <div
                  class="modal-dialog modal-dialog-centered"
                  role="document"
                >
                  <div
                    class="modal-content"
                  >
                    <div
                      class="modal-header"
                    >
                      <h4
                        class="modal-title"
                      >
                        Keyboard shortcuts
                      </h4>
                      <button
                        aria-label="Close"
                        class="btn btn-icon"
                        data-dismiss="modal"
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          ×
                        </span>
                      </button>
                    </div>
                    <div
                      class="modal-body"
                    >
                      <ul
                        class="list-group list-group-flush"
                      >
                        <li
                          class="list-group-item d-flex align-items-center justify-content-between"
                        >
                          t
                          <span>
                            <span>
                              <kbd>
                                Alt
                              </kbd>
                              <kbd>
                                x
                              </kbd>
                            </span>
                          </span>
                        </li>
                        <li
                          class="list-group-item d-flex align-items-center justify-content-between"
                        >
                          Expand URL to its canonical form (on file or tree page)
                          <span>
                            <span>
                              <kbd>
                                y
                              </kbd>
                            </span>
                          </span>
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="modal-backdrop show"
              />
            </div>
          </div>
        }
      >
        <div
          className=""
        >
          <Fade
            appear={true}
            baseClass=""
            baseClassActive="show"
            className="modal"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onClick={[Function]}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            role="dialog"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex="-1"
            tag="div"
            timeout={0}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={0}
              unmountOnExit={false}
            >
              <div
                className="modal"
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                role="dialog"
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex="-1"
              >
                <div
                  className="modal-dialog modal-dialog-centered"
                  role="document"
                >
                  <div
                    className="modal-content"
                  >
                    <div
                      className="modal-header"
                    >
                      <h4
                        className="modal-title"
                      >
                        Keyboard shortcuts
                      </h4>
                      <button
                        aria-label="Close"
                        className="btn btn-icon"
                        data-dismiss="modal"
                        onClick={[Function]}
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          ×
                        </span>
                      </button>
                    </div>
                    <div
                      className="modal-body"
                    >
                      <ul
                        className="list-group list-group-flush"
                      >
                        <li
                          className="list-group-item d-flex align-items-center justify-content-between"
                          key="x"
                        >
                          t
                          <span>
                            <span
                              key="0"
                            >
                              <kbd
                                key="0"
                              >
                                Alt
                              </kbd>
                              <kbd
                                key="1"
                              >
                                x
                              </kbd>
                            </span>
                          </span>
                        </li>
                        <li
                          className="list-group-item d-flex align-items-center justify-content-between"
                          key="canonicalURL"
                        >
                          Expand URL to its canonical form (on file or tree page)
                          <span>
                            <span
                              key="0"
                            >
                              <kbd
                                key="0"
                              >
                                y
                              </kbd>
                            </span>
                          </span>
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </div>
            </Transition>
          </Fade>
          <div
            className="modal-backdrop show"
          />
        </div>
      </Portal>
    </Portal>
  </Modal>
</KeyboardShortcutsHelp>
`;
